src/electric/adapter.ts

```diff
@@ -1,19 +1,22 @@
 import { AnyDatabase } from '../drivers/index'
 import { QualifiedTablename } from '../util/tablename'
-import { BindParams, Query, Row } from '../util/types'
+import { Row, Statement } from '../util/types'
 
 // A `DatabaseAdapter` adapts a database client to provide the
 // normalised interface defined here.
 export interface DatabaseAdapter {
   db: AnyDatabase
 
-  // Runs the provided sql.
-  run(sql: string): Promise<void>
+  // Runs the provided sql statement
+  run(statement: Statement): Promise<void>
+
+  // Runs the provided sql as a transaction
+  runTransaction(...statements: Statement[]): Promise<void>
```

Comment:
Nit but can I suggest `runInTransaction` -- just because you're passing in a list of statements, so you're running these in a transaction, as opposed to having a transaction around that you're running.